File: articles/angela-duckworth-and-the-modern-science-of-grit/c/duckworth_grit_score.c
#include <stdio.h>

double grit_score(double perseverance_effort, double consistency_interest) {
    return 0.60 * perseverance_effort + 0.40 * consistency_interest;
}

int main(void) {
    double perseverance_effort = 0.82;
    double consistency_interest = 0.61;
    printf("Duckworth-style synthetic grit score: %.3f\n",
           grit_score(perseverance_effort, consistency_interest));
    return 0;
}

File: articles/angela-duckworth-and-the-modern-science-of-grit/fortran/duckworth_grit_score.f90
program duckworth_grit_score
  implicit none
  real :: perseverance_effort, consistency_interest, grit_score

  perseverance_effort = 0.82
  consistency_interest = 0.61
  grit_score = 0.60 * perseverance_effort + 0.40 * consistency_interest

  print *, "Duckworth-style synthetic grit score:", grit_score
end program duckworth_grit_score
